src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,55 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+
+        try {
+            totalSupply = readData(fromFileName, SUPPLY);
+            totalBuy = readData(fromFileName, BUY);
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading data from file: " + fromFileName, e);
+        }
+
+        int result = totalSupply - totalBuy;
+
+        try (FileWriter writer = new FileWriter(toFileName)) {
+            writer.write(SUPPLY + "," + totalSupply + System.lineSeparator());
+            writer.write(BUY + "," + totalBuy + System.lineSeparator());
+            writer.write("result," + result + System.lineSeparator());
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing data to file: " + toFileName, e);
+        }
+    }
```

Comment:
This method is quite long and does multiple things. It reads data from a file, calculates the result, and writes data to another file. According to the 'Single Responsibility Principle', a method should have only one reason to change. Consider splitting this method into smaller ones, each with a single responsibility.